diff --git a/src/React.lua b/src/React.lua
index 23f94f2c..5366340b 100644
--- a/src/React.lua
+++ b/src/React.lua
@@ -102,6 +102,7 @@ return {
 	useLayoutEffect = ReactHooks.useLayoutEffect,
 	useMemo = ReactHooks.useMemo,
 	useMutableSource = ReactHooks.useMutableSource,
+	useSyncExternalStore = ReactHooks.useSyncExternalStore,
 	useReducer = ReactHooks.useReducer,
 	useRef = ReactHooks.useRef,
 	-- ROBLOX deviation: bindings support
diff --git a/src/ReactHooks.lua b/src/ReactHooks.lua
index 1a7dae7a..2b46bcb7 100644
--- a/src/ReactHooks.lua
+++ b/src/ReactHooks.lua
@@ -321,4 +321,14 @@ exports.useMutableSource = function<Source, Snapshot>(
 	return dispatcher.useMutableSource(source, getSnapshot, subscribe)
 end
 
+-- ROBLOX upstream: https://github.com/facebook/react/blob/34aa5cfe0d9b6ec4667e02bf46ab34d83dfb2d6d/packages/react/src/ReactHooks.js#L189-L200
+exports.useSyncExternalStore = function<T>(
+	subscribe: (() -> ()) -> () -> (),
+	getSnapshot: () -> T,
+	getServerSnapshot: (() -> T)?
+): T
+	local dispatcher = resolveDispatcher()
+	return dispatcher.useSyncExternalStore(subscribe, getSnapshot, getServerSnapshot)
+end
+
 return exports
