diff --git a/out/RojoResolver.js b/out/RojoResolver.js
index 6a3ede2..97a872b 100644
--- a/out/RojoResolver.js
+++ b/out/RojoResolver.js
@@ -160,8 +160,8 @@ class RojoResolver {
         return resolver;
     }
     parseConfig(rojoConfigFilePath, doNotPush = false) {
-        const realPath = fs_extra_1.default.realpathSync(rojoConfigFilePath);
-        if (fs_extra_1.default.pathExistsSync(realPath)) {
+        if (fs_extra_1.default.pathExistsSync(rojoConfigFilePath)) {
+            const realPath = fs_extra_1.default.realpathSync(rojoConfigFilePath);
             let configJson;
             try {
                 configJson = JSON.parse(fs_extra_1.default.readFileSync(realPath).toString());
@@ -196,6 +196,10 @@ class RojoResolver {
     }
     parsePath(itemPath) {
         itemPath = convertToLuau(itemPath);
+        if (ROJO_FILE_REGEX.test(path_1.default.basename(itemPath)) && fs_extra_1.default.pathExistsSync(itemPath)) {
+            this.parseConfig(itemPath, true);
+            return;
+        }
         const realPath = fs_extra_1.default.pathExistsSync(itemPath) ? fs_extra_1.default.realpathSync(itemPath) : itemPath;
         const ext = path_1.default.extname(itemPath);
         if (ROJO_MODULE_EXTS.has(ext)) {
